go-controller/pkg/kubevirt/dhcp.go

```diff
@@ -0,0 +1,138 @@
+package kubevirt
+
+import (
+	"fmt"
+	"net"
+
+	corev1 "k8s.io/api/core/v1"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	utilnet "k8s.io/utils/net"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+const (
+	dhcpLeaseTime = 3500
+)
+
+type dhcpConfigs struct {
+	V4 *nbdb.DHCPOptions
+	V6 *nbdb.DHCPOptions
+}
+
+func EnsureDHCPOptionsForVM(controllerName string, nbClient libovsdbclient.Client, watchFactory *factory.WatchFactory, vmKey ktypes.NamespacedName, ips []*net.IPNet, lsp *nbdb.LogicalSwitchPort) error {
+	dhcpConfigs, err := composeDHCPConfigs(watchFactory, controllerName, vmKey.Namespace, vmKey.Name, ips)
+	if err != nil {
+		return fmt.Errorf("failed composing DHCP options: %v", err)
+	}
+	err = libovsdbops.CreateOrUpdateDhcpOptions(nbClient, lsp, dhcpConfigs.V4, dhcpConfigs.V6)
+	if err != nil {
+		return fmt.Errorf("failed creation or updating OVN operations to add DHCP options: %v", err)
+	}
+	return nil
+}
+
+func composeDHCPConfigs(k8scli *factory.WatchFactory, controllerName, namespace, vmName string, podIPs []*net.IPNet) (*dhcpConfigs, error) {
```

Comment:
you can pass the namespaced name here (and pass it along the composeV4 and composeV6 funcs).

You split it in this func caller, then compose it again within those helpers.